src/CommentForm.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import { postComment } from "./app";
+
+const CommentForm = ({ article_id, comments, setComments }) => {
+  const [newComment, setNewComment] = useState({ username: "", body: "" });
+  const [addingComment, setAddingComment] = useState(false);
+  const [hasAddedComment, setHasAddedComment] = useState(false);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setNewComment({ ...newComment, [name]: value });
+  };
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    setAddingComment(true);
+    setHasAddedComment(false);
+    postComment(article_id, newComment).then((comment) => {
+      setAddingComment(false);
+      setHasAddedComment(true);
+      setComments([comment, ...comments]);
+      setNewComment({ username: "", body: "" });
+    });
+  };
```

Comment:
Nice logic, but what should happen if this request fails? Currently it just stays stuck on posting comment